src/borg/remote.py

```diff
@@ -19,6 +21,29 @@
 from .helpers import bin_to_hex
 from .repository import Repository
 
+if sys.platform == 'win32':
+    import ctypes
+    import ctypes.wintypes
+    import msvcrt
+    PeekNamedPipe = ctypes.windll.kernel32.PeekNamedPipe
+    PeekNamedPipe.argtypes = [ctypes.wintypes.HANDLE, ctypes.c_void_p, ctypes.wintypes.DWORD,
+        ctypes.POINTER(ctypes.wintypes.DWORD), ctypes.POINTER(ctypes.wintypes.DWORD),
+        ctypes.POINTER(ctypes.wintypes.DWORD)]
+    PeekNamedPipe.restype = ctypes.c_bool
+
+    def newSelect(rlist, wlist, xlist, timeout=0):
+        retRlist = []
+        retXlist = []
+        for pipe in rlist:
+            size = ctypes.wintypes.DWORD(0)
+            if not PeekNamedPipe(msvcrt.get_osfhandle(pipe), None, 0, None, ctypes.byref(size), None):
+                if size.value == 0 and pipe in xlist:
+                    retXlist.append(pipe)
+            if size.value > 0:
+                retRlist.append(pipe)
+        return retRlist, wlist, retXlist
+    select.select = newSelect
```

Comment:
select is part of WSA which only works for sockets and never has / never will work for anything else.